### PR TITLE
Fix build for wasm32 target

### PR DESCRIPTION
**Pull request by VincentFTS**
**Connections**
#115 

**Description**
Fix Wgpu build for Wasm target

**Testing**
It’s just a first step toward wasm support in Wgpu-rs, a PR in Wgpu-rs will come later.

### CODE REVIEW

File: wgpu-core/Cargo.toml [hunk 1]
```diff
@@ -39,19 +39,22 @@ gpu-descriptor = { git = "https://github.com/zakarumych/gpu-descriptor", rev = "
...
 gfx-backend-vulkan = { git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a", optional = true }
 
-[target.'cfg(windows)'.dependencies]
+[target.'cfg(all(not(target_arch = "wasm32"), windows))'.dependencies]
 gfx-backend-dx12 = { git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a" }
 gfx-backend-dx11 = { git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a" }
 gfx-backend-vulkan = { git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a" }
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+gfx-backend-gl = { git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a", features = ["naga"] }
```
Comment (kvark) [inline wgpu-core/Cargo.toml:56]: can we merge this with the other declaration?
```toml
[target.'cfg(any(target_arch = "wasm32", all(unix, not(target_os = "ios"), not(target_os = "macos"))))'.dependencies]
gfx-backend-gl = { git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a", features = ["naga"] }
```

File: wgpu-core/Cargo.toml [hunk 2]
```diff
@@ -39,19 +39,22 @@ gpu-descriptor = { git = "https://github.com/zakarumych/gpu-descriptor", rev = "
 hal = { package = "gfx-hal", git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a" }
 gfx-backend-empty = { git = "https://github.com/gfx-rs/gfx", rev = "0244e3401e9f127617cb8636397048584e7bfe8a" }
 
-[target.'cfg(all(unix, not(target_os = "ios"), not(target_os = "macos")))'.dependencies]
+[target.'cfg(all(not(target_arch = "wasm32"), all(unix, not(target_os = "ios"), not(target_os = "macos"))))'.dependencies]
```
Comment (kvark) [inline wgpu-core/Cargo.toml:42]: it's already `all(a, b)`, so we should just make it `all(a, b, c)` instead of wrapping everything into another `all`

**Review by kvark — COMMENTED** [review]
thank you for the fix!
made a few notes, also rustfmt is needed (and a squash afterwards)

File: wgpu-core/Cargo.toml (hunk 2 above)
Comment (VincentFTS) [reply wgpu-core/Cargo.toml:42]: My goal is to make this more readable by having the exact same pattern for the different backend dependency declarations.
Here again, let me know if you still want me to merge the two `all()`.

File: wgpu-core/Cargo.toml (hunk 1 above)
Comment (VincentFTS) [reply wgpu-core/Cargo.toml:56]: The dependencies seems to be classified by target, that’s why I isolated Wasm one.
Let me know if you still want me to merge this with the other declaration.

**Comment by VincentFTS** [comment]
`cargo fmt` is clear here.
I guess you mean `cargo clippy` which was already quite verbose on master ;)

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 /target
+*/target
```
Comment (kvark) [inline .gitignore:2]: what's the path this is trying to include, exactly? we could do just "target"  to combine the two

**Review by kvark — APPROVED** [review]
bors r+

**Comment by kvark** [comment]
bors r-
actually, we need a CI target for the wasm build now

File: .gitignore (hunk 3 above)
Comment (VincentFTS) [reply .gitignore:2]: Wgpu-core/target and Wgpu-types/target

**Review by kvark — APPROVED** [review]
bors r+

**Pull request merged**